design/proposals/variable_provision.md

```diff
@@ -0,0 +1,295 @@
+|          |                 |
+
+## Useful links
+[//]: # (Add links that may be useful for the reader)
+
+> TBD
+
+## Background
+
+The purpose of Variables in Conjur is to put values in them for applications to
+use for service access. These values often consist of:
+
+- Passwords
+- API Keys
+- x509 Private Keys
+- x509 Certificates
+
+The typical workflow for a Variable is:
+
+- The Variable is created when it is defined in a Conjur policy document and loaded
+  into Conjur:
+
+  ```YAML
+  # my-variable.yaml
+
+  - !variable my-api-key
+  ```
+
+  ```sh-session
+  $ conjur policy load root my-variable.yaml
+  ...
+  ```
+
+- The owner of the value uses the Conjur API or CLI to store the value in the Variable:
+  
+  ```sh-session
+  $ conjur variable values add my-api-key abcdef12345
+  ...
+  ```
+
+- Once stored, access to the variable may be granted to applications, and those
+  applications can use value using the Conjur API, Summon, Secretless, etc.
+
+## Issue description
+
+There are a couple of issues this solution intends to address:
+
+### 'Browsing-up` for secrets management
+
+"Browse-up" is defined as [[ref]](https://www.ncsc.gov.uk/whitepaper/security-architecture-anti-patterns#section_3):
+
+> When administration of a system is performed from a device which is less trusted
+> than the system being administered.
+
+Unless performed only on dedicated, controlled workspaces, the `variable values add`
+step in our existing workflow necessitates that all secrets stored in Conjur (a
+"trusted system") must first travel through a user's workstation (a "less trusted system").
+
+### Secrets Domain Knowledge Required
+
+This isn't an "issue" as much as an opportunity for added value. Conjur today serves
+essentially as a database with encryption and audit. While these are crucial, as
+a Secrets store and Access Control system, there is an opportunity to provide more
+access control domain knowledge in Conjur itself, rather than requiring this as
+pre-requisite knowledge a user needs.
+
+An example of this is X.509 Certificates used for authentication, secure communications,
+and data integrity. While it may be simple enough to understand why I would want
+to use Certificates in my applications and infrastructure. Creating and maintaining
+these certificates, and a public key infastructure (PKI) is
+[anything but trivial](https://smallstep.com/blog/everything-pki/). And mistakes
+here can invalidate the security gains from using certificates in the first place.
+
+### Human Intervention Required
+
+As application deployments scale up, the need to automate the creation and rotation
+of credentials will continue to grow as human provisioning of credentails becomes
+a bottleneck. Our API enables end-users to bolt this automation onto Conjur, but
+we have an opportunity to enable more just-in-time creation of these credentails
+on behalf the user to make this kind of automation more widely available.
+
+### Credential Provenance
```

Comment:
This really struck me.  I've never thought about the fact that these credentials likely live an entire life before they come into the DAP/Conjur world. During that time, no one really knows where they are with complete certainty.